pallets/ah-migrator/src/staking/nom_pools.rs

```diff
@@ -0,0 +1,139 @@
+			},
+			PoolMembers { member } => {
+				debug_assert!(!pallet_nomination_pools::PoolMembers::<T>::contains_key(&member.0));
+				log::debug!("Integrating NomPoolsPoolMember: {:?}", &member.0);
+				pallet_nomination_pools::PoolMembers::<T>::insert(member.0, member.1);
+			},
+			BondedPools { pool } => {
+				debug_assert!(!pallet_nomination_pools::BondedPools::<T>::contains_key(&pool.0));
+				log::debug!("Integrating NomPoolsBondedPool: {}", &pool.0);
+				pallet_nomination_pools::BondedPools::<T>::insert(
+					pool.0,
+					Self::rc_to_ah_bonded_pool(pool.1),
+				);
+			},
+			RewardPools { rewards } => {
+				log::debug!("Integrating NomPoolsRewardPool: {:?}", &rewards.0);
+				// Not sure if it is the best to use the alias here, but it is the easiest...
+				pallet_rc_migrator::staking::nom_pools_alias::RewardPools::<T>::insert(
+					rewards.0, rewards.1,
+				);
+			},
+			SubPoolsStorage { sub_pools } => {
+				log::debug!("Integrating NomPoolsSubPoolsStorage: {:?}", &sub_pools.0);
+				pallet_rc_migrator::staking::nom_pools_alias::SubPoolsStorage::<T>::insert(
+					sub_pools.0,
+					sub_pools.1,
+				);
+			},
+			Metadata { meta } => {
+				log::debug!("Integrating NomPoolsMetadata: {:?}", &meta.0);
+				pallet_nomination_pools::Metadata::<T>::insert(meta.0, meta.1);
+			},
+			ReversePoolIdLookup { lookups } => {
+				log::debug!("Integrating NomPoolsReversePoolIdLookup: {:?}", &lookups.0);
+				pallet_nomination_pools::ReversePoolIdLookup::<T>::insert(lookups.0, lookups.1);
+			},
+			ClaimPermissions { perms } => {
+				log::debug!("Integrating NomPoolsClaimPermissions: {:?}", &perms.0);
+				pallet_nomination_pools::ClaimPermissions::<T>::insert(perms.0, perms.1);
+			},
+		}
+	}
+
+	/// Translate a bonded RC pool to an AH one.
+	pub fn rc_to_ah_bonded_pool(mut pool: BondedPoolInner<T>) -> BondedPoolInner<T> {
+		if let Some(ref mut throttle_from) = pool.commission.throttle_from {
+			// Plus one here to be safe for the pool member just in case that the pool operator
+			// would like to enact commission rate changes immediately.
+			*throttle_from = Self::rc_to_ah_timepoint(*throttle_from).saturating_add(One::one());
+		}
+		if let Some(ref mut change_rate) = pool.commission.change_rate {
+			// We cannot assume how this conversion works, but adding one will ensure that we err on
+			// the side of pool-member safety in case of rounding.
+			change_rate.min_delay =
+				T::RcToAhDelay::convert(change_rate.min_delay).saturating_add(One::one());
+		}
+
+		pool
+	}
+
+	/// Convert an absolute RC time point to an AH one.
+	///
+	/// This works by re-anchoring the time point to. For example:
+	///
+	/// - RC now: 100
+	/// - AH now: 75
+	/// - RC time point: 50
+	/// - Result: 75 - (100 - 50) / 2 = 50
+	///
+	/// Other example:
+	///
+	/// - RC now: 100
+	/// - AH now: 75
+	/// - RC time point: 150
+	/// - Result: 75 + (150 - 100) / 2 = 100
+	pub fn rc_to_ah_timepoint(rc_timepoint: BlockNumberFor<T>) -> BlockNumberFor<T> {
```

Comment:
for all real time sensitive timepoints I am gonna leave RC as a block provider and do not do the mapping. If it just for servicing, my plan is to keep the system block provider.

Reply:
The nom pools pallet is not migrated to RC number provider yet, otherwise i would also prefer that.

Reply:
it is in the list, so yes we did plan it 
https://github.com/paritytech/polkadot-sdk/issues/6297

Reply:
Would be better to have it. But turning off the translation here is a two line change so am not too concerned about it.